primitives/inherents/src/lib.rs

```diff
@@ -408,6 +408,11 @@ pub trait ProvideInherent {
 	/// Create an inherent out of the given `InherentData`.
 	fn create_inherent(data: &InherentData) -> Option<Self::Call>;
 
+	/// If `Some`, indicates that an inherent is required. Check will return the inner error if no
+	/// inherent is found. If `Err`, indicates that the check failed and further operations should
+	/// be aborted.
+	fn is_inherent_required(_: &InherentData) -> Result<Option<Self::Error>, Self::Error> { Ok(None) }
```

Comment:
```suggestion
	fn is_inherent_required(_: &InherentData) -> Option<Self::Error> { None }
```

Reply:
The `Err` type and `Some` type has different meanings.

* Calling this function, the procedure can fail. For example, this can due to decoding of `InherentData` failed or runtime storage value fetch failed. In this case, `Err` is returned.
* If calling this function succeeds and the runtime determines that an inherent is indeed required, then `Some` is returned. The error will be returned if later runtime checks that the inherent does not exist.

Reply:
Yeah I understand the difference, however from the caller it makes no difference. The caller will abort if there is any error.

Reply:
There's actually some difference -- if it returns `Err`, the caller would immediately aborts. However, if that is wrapped in `Some`, then the error would be used only if an inherent is not found, which means if the inherent is found, then the function will actually succeed!

I think the main point is -- while `Err` is an actual error, the thing wrapped in `Option` is not an actual error. The later only becomes an actual error if an inherent is not found, otherwise it only indicates that an inherent is required. Not sure if there are better ways to make this clearer.